### PR TITLE
Use the session.transaction context manager for nicer exceptions

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: luigi/db_task_history.py [hunk 1]
```diff
@@ -65,13 +65,8 @@ def _session(self, session=None):
             yield session
         else:
             session = self.session_factory()
-            try:
+            with session.transaction:
```
Comment (graingert) [inline luigi/db_task_history.py]: SQLAlchemy's session management does magic to make sure that if the rollback fails you still get the original exception that caused the rollback.

Also it looks nicer.

**Comment by Tarrasch** [comment]
LGTM

**Comment by Tarrasch** [comment]
I'm not sure of the test coverage of the task history as I don't use it. Have you either tested this in production or do we have unit tests covering your change @graingert? If so I can merge.

**Comment by Tarrasch** [comment]
Closing as it's been stale for a while. Please reopen when you either have a test or at least verified it's running in production.

**Pull request closed without merging**